src/common/debounce.ts

```diff
@@ -1,10 +1,9 @@
-export function debounce(fn: (any) => any, ms: number): () => void {
-    let timer
-    return () => {
-        clearTimeout(timer)
-        timer = setTimeout((...args) => {
-            timer = null
-            fn.apply(this, ...args)
-        }, ms)
-    }
-}
+export const debounce =
+  <F extends (...args: Parameters<F>) => ReturnType<F>>(fn: F, ms: number): (...args: Parameters<F>) => void => {
```

Comment:
Should we use `function` instead of arrow function here? The `=> void => {}` confuses where is the return type definition ends and when is the arrow function starting